src/geometry/SE3vel.cpp

```diff
@@ -0,0 +1,200 @@
+#include "mrob/SE3vel.hpp"
+
+using namespace mrob;
+
+SE3vel::SE3vel(const Mat5 &T) : T_(T) {}
+
+SE3vel::SE3vel(const SE3vel &T) : T_(T.T()){}
+
+SE3vel::SE3vel(const Mat91 &xi)
+{
+    this->Exp(xi);
+}
+ 
+Mat31 SE3vel::t() const
+{
+    return T_.topRightCorner<3,1>();
+}
+
+Mat31 SE3vel::v() const
+{
+    return T_.block<3,1>(0,3);
+}
+
+Mat3 SE3vel::R() const
+{
+    return T_.topLeftCorner<3,3>();
+}
+
+Mat5 SE3vel::T(void) const
+{
+    return this->T_;
+}
+
+SE3vel SE3vel::inv(void) const
+{
+    Mat5 inv(Mat5::Zero());
+    Mat3 R = this->R();
+    R.transposeInPlace();
+
+    inv << R, -R*this->v(), -R*this->t(),
+            0,0,0,1,0,
+            0,0,0,0,1;
+
+    return SE3vel(inv);
+}
+
+SE3vel operator*(const SE3vel& lhs, const SE3vel& rhs)
+{
+    return SE3vel(Mat5(lhs.T()*rhs.T()));
+}
+
+Mat9 SE3vel::adj() const
+{
+    Mat9 res(Mat9::Zero());
+    Mat3 R = this->R();
+    Mat31 v = this->v();
+    Mat31 t = this->t();
+
+    res.block<3,3>(0,0) = R;
+    res.block<3,3>(3,3) = R;
+    res.block<3,3>(6,6) = R;
+
+    res.block<3,3>(3,0) = hat3(v)*R;
+    res.block<3,3>(6,0) = hat3(t)*R;
+
+    return res;
+}
+
+std::ostream& operator<<(std::ostream &os, const SE3vel& obj)
+    {
+        os << obj.T();
+        return os;
+    }
+
+
+Mat5 hat9(const Mat91 &xi)
+{
+    Mat5 result(Mat5::Zero());
+
+    result <<    0, -xi(2),  xi(1), xi(3), xi(6),
+             xi(2),      0, -xi(0), xi(4), xi(7),
+            -xi(1),  xi(0),      0, xi(5), xi(8),
+                 0,      0,      0,     0,     0,
+                 0,      0,      0,     0,     0;
+
+    return result;   
+}
+
+
+/**
+ * Vee operator (v), the inverse of hat
+ */
+Mat91 vee9(const Mat4 &xi_hat)
+{
+    Mat91 result(Mat91::Zero());
+
+    result << xi_hat(2,1), xi_hat(0,2), xi_hat(1,0),
+              xi_hat(0,3), xi_hat(1,3), xi_hat(2,3),
+              xi_hat(0,4), xi_hat(1,4), xi_hat(2,4);
+
+    return result;
+}
+
+
+Mat3 SE3vel::left_jacobian(const Mat31& phi)
```

Comment:
is this not inside SO3? If that is the case, should it be added directly there? I am ok adding an issue and changing SO3